File: Hobby-hub/Data Models/Hobby.cs
using Hobby_hub.Data_Models.Enum;

namespace Hobby_hub.Data_Models
{
    public class Hobby
    {
        public int Id { get; set; }

        public string Name { get; set; } = string.Empty;

        public string Description { get; set; } = string.Empty;

        public HobbyScores Scores { get; set; } = new();

        public string YoutubeVideoId { get; set; } = string.Empty;

        public string HobbyImage {  get; set; } = string.Empty;

        public List<Category> HobbyCategory { get; set; }  
    }

}

File: hobby-hub-frontend/src/models/hobby.ts
export type Hobby = {
    id: number;
    name: string;
    description: string;
    longDescription: string;
    scores: HobbyScores;
    youtubeVideoId: string;
    hobbyImage: string;
    hobbyImageAltText: string;
    hobbyCategory: Category[];
}

export type HobbyScores = {
    active: number;
    creative: number;
    relaxing: number;
    social: number;
    outdoor: number;
    strategic: number;
    price: number;
    timeCommitment: number;
}

export type Category = "Active" | "Creative" | "Relaxing" | "Social" | "Outdoor" | "Strategic";